Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift

```diff
@@ -112,9 +112,7 @@ public struct MessageContainerView<Factory: ViewFactory>: View {
                         GeometryReader { proxy in
                             Rectangle().fill(Color.clear)
                                 .onChange(of: computeFrame, perform: { _ in
-                                    DispatchQueue.main.async {
-                                        frame = proxy.frame(in: .global)
-                                    }
+                                    frame = proxy.frame(in: .global)
```

Comment:
Already on main thread and onChange is called just after SwiftUI finishes updating views, therefore should be safer to ask for the frame than delaying it. Not sure if it fixes the reported crash since I can't reproduce it, but worth the try.